llvm/lib/Transforms/Vectorize/LoopVectorizationPlanner.h

```diff
@@ -66,6 +67,12 @@ class VPBuilder {
     return createInstruction(Opcode, ArrayRef<VPValue *>(Operands), DL, Name);
   }
 
+  VPValue *getOrAddLiveIn(Value *V) {
+    if (BB)
+      return BB->getPlan()->getOrAddLiveIn(V);
+    return new VPValue(V);
```

Comment:
This will introduce a leak?

Reply:
Will it? I thought the Builder is used in two different ways: when the insertion point is set, and when the insertion point isn't set. For the case when the insertion point isn't set, wouldn't the following introduce a leak, if this introduces a leak?

```cpp
template <typename T> T *tryInsertInstruction(T *R) {
    if (BB)
      BB->insert(R, InsertPt);
    return R;
  }
  VPInstruction *createInstruction(unsigned Opcode,
                                   ArrayRef<VPValue *> Operands, DebugLoc DL,
                                   const Twine &Name = "") {
    return tryInsertInstruction(new VPInstruction(Opcode, Operands, DL, Name));
  }
```